eodatasets3/scripts/tostac.py

```diff
@@ -7,70 +7,182 @@
-            id=dataset.id,
-            type="Feature",
-            bbox=wgs84_geometry.bounds,
-            geometry=wgs84_geometry.__geo_interface__,
-            properties={**dataset.properties, "odc:product": dataset.product.name},
-            assets={
-                # TODO: Currently assuming no name collisions.
-                **{name: {"href": m.path} for name, m in dataset.measurements.items()},
-                **{name: {"href": m.path} for name, m in dataset.accessories.items()},
-            },
-            links=[
-                # {
-                #     "rel": "self",
-                #     "href": '?',
-                # },
-                {"rel": "odc_product", "href": dataset.product.href},
-                {
-                    "rel": "alternative",
-                    "type": "text/html",
-                    "href": f"https://explorer.dea.ga.gov.au/dataset/{dataset.id}",
-                },
-            ],
-        )
-
         name = input_metadata.stem.replace(".odc-metadata", "")
         output_path = input_metadata.with_name(f"{name}.stac-item.json")
 
+        # Load STAC template data
+        if stac_template:
+            with open(stac_template, "r") as f:
+                stac_data = json.load(f)
+        else:
+            stac_data = {}
+
+        # Create STAC dict
+        item_doc = create_stac(
+            dataset,
+            input_metadata,
+            output_path,
+            stac_data,
+            stac_base_url,
+            product_base_url,
+        )
+
         with output_path.open("w") as f:
             json.dump(item_doc, f, indent=4, default=json_fallback)
 
-        echo(output_path)
+
+def create_stac(
+    dataset, input_metadata, output_path, stac_data, stac_base_url, product_base_url
+):
+    """
+    Create STAC document
+
+    :param dataset: Dict of the metadata content
+    :param input_metadata: Path of the Input metadata file
+    :param output_path: Path of the STAC output file
+    :param stac_data: Dict of the static STAC content
+    :param stac_base_url: Base URL for STAC file
+    :param product_base_url: Base URL for ODC product
+    :return: Dict of the STAC content
+    """
+
+    project = partial(
+        pyproj.transform, pyproj.Proj(init=dataset.crs), pyproj.Proj(init="epsg:4326"),
+    )
+    wgs84_geometry: BaseGeometry = transform(project, dataset.geometry)
+    item_doc = dict(
+        stac_version=stac_data.get("stac_version", "1.0.0-beta.1"),
+        stac_extensions=stac_data.get("stac_extensions", []),
+        id=dataset.id,
+        type="Feature",
+        bbox=wgs84_geometry.bounds,
+        geometry=wgs84_geometry.__geo_interface__,
+        properties={
+            **{
+                MAPPING_EO3_TO_STAC.get(key, key): convert_value_to_stac_type(key, val)
+                for key, val in dataset.properties.items()
+            },
+            "odc:product": dataset.product.name,
+            "proj:epsg": dataset.crs.lstrip("epsg:"),
```

Comment:
This might fail badly for non-epsg definitions. In the short term we can just throw a non-implemented exception for those (I prefer that to silently corrupt results)

Reply:
I agree. Just including the numbers here is enough, splitting on `:` is hopefully safe...

It's a problem with the `proj` extension though, as assuming `epsg` as the prefix is not right.

Reply:
It looks like in the spec the field is mandatory, but the value can be null, in which case one of the others like `proj:wkt2` is specified for non-epsg.

Reply:
`proj:epsg` value safely converted to `int` else set `None`.